src/dawn/Optimizer/PassSetBoundaryCondition.cpp

```diff
@@ -91,8 +91,12 @@ class AddBoundaryConditions : public ASTVisitorForwarding {
       : instantiation_(instantiation), StencilID_(StencilID) {}
 
   void visit(const std::shared_ptr<StencilCallDeclStmt>& stmt) override {
-    if(instantiation_->getStencilIDFromStmt(stmt) == StencilID_)
-      stencilCallsToReplace_.emplace_back(stmt);
+    auto it = instantiation_->getStencilCallToStencilIDMap().find(stmt);
```

Comment:
how about using a find_if with a lambda?
What happens if not found. Is it a failure, should we assert or something?

Reply:
nothing should happen. The visitor is called on every stmt in the StencilDescAST. If no stencil to be replaced is found, replacement is skipped and we move on